.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,51 @@
+name: Build and Deploy to Netlify
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Install JupyterLite CLI
+        run: pip install jupyterlite
```

Comment:
The dependencies are already specified in `requirements.txt`. This explicit installation step shouldn't be required. If it is required, it should be added to `requirements.txt`.